pkg/scheduler/core/equivalence_cache.go

```diff
@@ -86,6 +87,12 @@ func (ec *EquivalenceCache) RunPredicate(
 ) (bool, []algorithm.PredicateFailureReason, error) {
 	ec.mu.Lock()
 	defer ec.mu.Unlock()
+
+	if nodeInfo == nil || nodeInfo.Node() == nil {
+		// This may happen during tests.
+		return false, []algorithm.PredicateFailureReason{}, fmt.Errorf("nodeInfo is nil or node is invalid")
```

Comment:
I think this should return false, nil, fmt.Errorf(...), but I know that breaks tests today.